File: internal/model/account.go
package model

import (
	"encoding/json"
	"time"
)

// WechatAccount 微信公众号
// 对应表: wechat_accounts
type WechatAccount struct {
	ID             int64      `db:"id"`
	TenantID       int64      `db:"tenant_id"`
	Name           *string    `db:"name"`
	WxOriginalID   *string    `db:"wx_original_id"`
	WxAppID        *string    `db:"wx_app_id"`
	WxAppSecret    *string    `db:"wx_app_secret"`
	AuthType       int16      `db:"auth_type"`
	AuthStatus     int16      `db:"auth_status"`
	RefreshToken   *string    `db:"refresh_token"`
	AccessToken    *string    `db:"access_token"`
	TokenExpireAt  *time.Time `db:"token_expire_at"`
	AvatarURL      *string    `db:"avatar_url"`
	QRCodeURL      *string    `db:"qr_code_url"`
	Description    *string    `db:"description"`
	FansCount      int        `db:"fans_count"`

	// T3 新增字段
	AuthorizerAppid  *string           `db:"authorizer_appid"`
	FuncInfo         *json.RawMessage  `db:"func_info"`          // 设计修正: *json.RawMessage
	ServiceTypeInfo  *int16            `db:"service_type_info"`
	VerifyTypeInfo   *int16            `db:"verify_type_info"`
	NickName         *string           `db:"nick_name"`
	HeadImg          *string           `db:"head_img"`
	UserName         *string           `db:"user_name"`
	Alias            *string           `db:"alias"`
	PrincipalName    *string           `db:"principal_name"`
	QrcodeUrl        *string           `db:"qrcode_url"`         // 微信同步的二维码（复用已有 qr_code_url）
	Signature        *string           `db:"signature"`

	DeletedAt      *time.Time `db:"deleted_at"`
	CreatedAt      time.Time  `db:"created_at"`
	UpdatedAt      time.Time  `db:"updated_at"`
}

File: web/admin/src/api/account.ts
import client from './client';

export interface CreateAccountParams {
  name: string;
  wx_app_id: string;
  wx_app_secret: string;
  wx_original_id?: string;
  description?: string;
  avatar_url?: string;
  qr_code_url?: string;
}

export interface UpdateAccountParams {
  name?: string;
  wx_app_id?: string;
  wx_app_secret?: string;
  wx_original_id?: string;
  description?: string;
  avatar_url?: string;
  qr_code_url?: string;
}

export interface AccountListParams {
  page?: number;
  page_size?: number;
  keyword?: string;
  auth_type?: number;
  auth_status?: number;
}

export interface AccountVO {
  id: number;
  tenant_id: number;
  name: string;
  wx_app_id: string;
  wx_original_id?: string;
  auth_type: number;
  auth_status: number;
  description?: string;
  avatar_url?: string;
  qr_code_url?: string;
  fans_count: number;
  nick_name?: string;
  head_img?: string;
  principal_name?: string;
  token_expire_at?: string;
  created_at: string;
  updated_at: string;
}

export interface PaginatedResponse<T> {
  list: T[];
  total: number;
  page: number;
  page_size: number;
}

export interface ApiResponse<T> {
  code: number;
  msg: string;
  data: T;
}

// Create account (manual setup)
export function createAccount(data: CreateAccountParams): Promise<{ data: ApiResponse<AccountVO> }> {
  return client.post('/accounts', data);
}

// List accounts (paginated)
export function listAccounts(params: AccountListParams): Promise<{ data: ApiResponse<PaginatedResponse<AccountVO>> }> {
  return client.get('/accounts', { params });
}

// Get account detail
export function getAccount(id: number): Promise<{ data: ApiResponse<AccountVO> }> {
  return client.get(`/accounts/${id}`);
}

// Update account
export function updateAccount(id: number, data: UpdateAccountParams): Promise<{ data: ApiResponse<AccountVO> }> {
  return client.put(`/accounts/${id}`, data);
}

// Delete account (soft delete)
export function deleteAccount(id: number): Promise<{ data: ApiResponse<null> }> {
  return client.delete(`/accounts/${id}`);
}
